opba-facade-protocol-api-shared/src/main/java/de/adorsys/opba/protocol/api/dto/result/body/ReturnableProcessErrorResult.java

```diff
@@ -0,0 +1,14 @@
+package de.adorsys.opba.protocol.api.dto.result.body;
+
+import de.adorsys.opba.protocol.api.dto.result.fromprotocol.Result;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public class ReturnableProcessErrorResult<T> implements Result<T> {
+    /** the error code should be returned to the caller, it may be used in the
+     * rest layer to use a specific http response code
+     */
+    private final int errorCode;
```

Comment:
Why we use `int errorCode` and not an enum/DTO that contains both HttpStatus(int) and ErrorCode(string)

Reply:
instead of using raw int values we should use ENUM. The enum can reside in `opba-facade-protocol-api-shared` this is the directory where all shared classes should reside (globally shared)

```
enum ParseableErrorCode {
   ACCESS_RATE_EXCEEDED;
}
```
`ReturnableProcessErrorResult`:
private final int errorCode; -> private final ParseableErrorCode errorCode;

then REST layer does:
if (error.errorCode == ACCESS_RATE_EXCEEDED) {
   return HttpStatus.429;
}

Reply:
Does not look to me like layered architecture. Facade is above protocol. So how can protocol use anything of facade?

Reply:
The protocol is not using anything from Facade. It uses enum from `opba-facade-protocol-api-shared`. 
`REST`- depends on -> `opba-facade-protocol-api-shared` <- depends on - `protocol` like an API layer for these two.

The REST layer is simply checking 
```
if (error.errorCode == ACCESS_RATE_EXCEEDED) {
return HttpStatus.429;
}
```
where ACCESS_RATE_EXCEEDED is in `opba-facade-protocol-api-shared`